src/order/order-action-creator.ts

```diff
@@ -68,14 +68,25 @@ export default class OrderActionCreator {
                 const state = store.getState();
                 const externalSource = state.config.getExternalSource();
                 const checkout = state.checkout.getCheckout();
+                const orderMeta = state.order.getOrderMeta();
+                const storeConfig = state.config.getStoreConfig();
+                const spamProtectionToken = orderMeta && orderMeta.spamProtectionToken;
+
+                if (!storeConfig) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+                }
 
                 if (!checkout) {
                     throw new MissingDataError(MissingDataErrorType.MissingCheckout);
                 }
 
+                if (storeConfig.checkoutSettings.isSpamProtectionEnabled && !spamProtectionToken) {
+                    throw new MissingDataError(MissingDataErrorType.MissingSpamProtection);
+                }
+
                 return from(
                     this._checkoutValidator.validate(checkout, options)
-                        .then(() => this._orderRequestSender.submitOrder(this._mapToOrderRequestBody(payload, checkout.customerMessage, externalSource), options))
+                        .then(() => this._orderRequestSender.submitOrder(this._mapToOrderRequestBody(payload, checkout.customerMessage, externalSource, spamProtectionToken), options))
```

Comment:
this line it's already a bit long, might be good to split the arguments into new lines
```javascript
.then(() => this._orderRequestSender.submitOrder(this._mapToOrderRequestBody(
  payload,
  checkout.customerMessage,
  externalSource,
  spamProtectionToken,
), options))
```